### PR TITLE
Use buildah commit and bud in podman

### LINKED ISSUE #586: podman commit: does not create entrypoint?
```
# podman run --name foo fedora date
Tue Apr  3 15:05:07 UTC 2018
# podman commit --quiet foo myimg
# podman run myimg
No command specified on command line or as CMD or ENTRYPOINT in this image
```
Expected: date output.

podman-0.3.5-1.gitdb6bf9e.fc27.x86_64

### PR DESCRIPTION
**Pull request by baude**
Vendor in buildah and use as much of commit and bud as possible for podman
build and commit.

Resolves #586
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
LGTM

File: libpod/container_commit.go [hunk 1]
```diff
@@ -68,6 +68,39 @@ func (c *Container) Commit(ctx context.Context, destImage string, options Contai
...
 
+	// We need to take meta we find in the current container and
+	// add it to the resulting image.
+
+	// Entrypoint - always set this first or cmd will get wiped out
+	importBuilder.SetEntrypoint(c.Spec().Process.Args)
+	// Cmd
+	// We cannot differentiate between cmd and entrypoint here
+	// so we assign args to both
+	importBuilder.SetCmd(c.Spec().Process.Args)
```
Comment (nalind) [inline libpod/container_commit.go]: When we "docker commit" a container there, which of these is/are set in the image's configuration when you then "docker inspect" the result?

**Comment by rhatdan** [comment]
Can you grab the tests from buildah.bud to make sure they work.
bud.bats and all of the subdirs of bud.

**Comment by baude** [comment]
you want to add bats tests back into libpod?

**Comment by rhatdan** [comment]
@baude or convert them to the new format.

File: libpod/container_commit.go (hunk 1 above)
Comment (baude) [reply libpod/container_commit.go]: The both appear to make it

**Comment by TomSweeneyRedHat** [comment]
Travis tests are failing with a seccomp error that I don't recall seeing before....

```
# github.com/projectatomic/libpod/vendor/github.com/seccomp/libseccomp-golang
vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:25:2: error: #error Minimum supported version of Libseccomp is v2.2.0
 #error Minimum supported version of Libseccomp is v2.2.0
  ^
make: *** [podman] Error 2
```

File: cmd/podman/build.go [hunk 2]
```diff
@@ -1,270 +1,174 @@
...
-	if c.IsSet("cert-dir") {
-		budCmdArgs = append(budCmdArgs, "--cert-dir", c.String("cert-dir"))
-	}
-	if c.IsSet("creds") {
-		budCmdArgs = append(budCmdArgs, "--creds", c.String("creds"))
-	}
-	for _, fileName := range c.StringSlice("file") {
-		budCmdArgs = append(budCmdArgs, "--file", fileName)
-	}
+	// The following was taken directly from projectatomic/buildah/cmd/bud.go
```
[the remark this answers is not in the document]
Comment (baude) [reply cmd/podman/build.go]: i didnt see any obvious ways

File: cmd/podman/build.go [hunk 3]
```diff
@@ -1,270 +1,174 @@
...
+	tags := []string{}
+	if c.IsSet("tag") || c.IsSet("t") {
+		tags = c.StringSlice("tag")
+		if len(tags) > 0 {
+			output = tags[0]
+			tags = tags[1:]
+		}
+	}
+	pullPolicy := imagebuildah.PullNever
+	if c.BoolT("pull") {
```
[the remark this answers is not in the document]
Comment (baude) [reply cmd/podman/build.go]: if they are local variables yes.

File: cmd/podman/ioprojectatomicpodman/ioprojectatomicpodman.go [hunk 4]
```diff
@@ -1038,267 +1038,267 @@ func (c__ *VarlinkCall) ReplyStopContainer(notimplemented_ NotImplemented) error
...
 
 	case "AttachToContainer":
 		return s__.ioprojectatomicpodmanInterface.AttachToContainer(VarlinkCall{call})
 
-	case "TagImage":
-		return s__.ioprojectatomicpodmanInterface.TagImage(VarlinkCall{call})
-
-	case "ImportImage":
-		return s__.ioprojectatomicpodmanInterface.ImportImage(VarlinkCall{call})
+	case "PullImage":
```
[the remark this answers is not in the document]
Comment (baude) [reply cmd/podman/ioprojectatomicpodman/ioprojectatomicpodman.go]: that is a dynamically generated file.  nothing i can do about that

**Comment by mheon** [comment]
Some failing commit tests

File: cmd/podman/build.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rhatdan) [reply cmd/podman/build.go]: Lets move forward on this and revisit it in the future. @baude Could you add a FIXME or something suggesting that we look into a way to vendor or share the code between podman and buildah.

**Comment by mheon** [comment]
LGTM once tests are green

File: pkg/inspect/inspect.go [hunk 5]
```diff
@@ -125,6 +125,7 @@ type ImageData struct {
 	RootFS          *RootFS           `json:"RootFS"`
 	Labels          map[string]string `json:"Labels"`
 	Annotations     map[string]string `json:"Annotations"`
+	ManifestType    string            `json:ManifestType`
```
Comment (mheon) [inline pkg/inspect/inspect.go]: Need "" marks around ManifestType

Comment (baude) [reply pkg/inspect/inspect.go]: fixed

**Comment by mheon** [comment]
Need to manually build + install libseccomp >= 2.2.0 for travis

File: .travis.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@ before_install:
   - sudo apt-get -qq install autoconf automake bison e2fslibs-dev libfuse-dev libtool liblzma-dev gettext
 
 install:
-  - make install.tools
+  - make install.libseccomp.sudo install.tools
```
Comment (mheon) [inline .travis.yml]: The Travis tests are all containerized, so these both do nothing

Comment (mheon) [reply .travis.yml]: Need to add install.libseccomp.sudo to the Dockerfile and remove `libseccomp` from the things apt installs in the dockerfile

File: Makefile [hunk 7]
```diff
@@ -249,6 +249,13 @@ install.tools: .install.gitvalidation .install.gometalinter .install.md2man
 
 varlink_generate: .gopathok cmd/podman/varlink/ioprojectatomicpodman.go
 
+.PHONY: install.libseccomp.sudo
```
Comment (mheon) [inline Makefile]: Move this to global .PHONY section

File: Makefile [hunk 8]
```diff
@@ -249,6 +249,13 @@ install.tools: .install.gitvalidation .install.gometalinter .install.md2man
...
+install.libseccomp.sudo:
+	rm -rf ../../seccomp/libseccomp
+	git clone https://github.com/seccomp/libseccomp ../../seccomp/libseccomp
+	cd ../../seccomp/libseccomp && git checkout $(LIBSECCOMP_COMMIT) && ./autogen.sh && ./configure --prefix=/usr && make all && sudo make install
+
+
```
Comment (mheon) [inline Makefile]: Extra newline

File: Makefile [hunk 9]
```diff
@@ -249,6 +249,13 @@ install.tools: .install.gitvalidation .install.gometalinter .install.md2man
 
 varlink_generate: .gopathok cmd/podman/varlink/ioprojectatomicpodman.go
 
+.PHONY: install.libseccomp.sudo
+install.libseccomp.sudo:
+	rm -rf ../../seccomp/libseccomp
+	git clone https://github.com/seccomp/libseccomp ../../seccomp/libseccomp
+	cd ../../seccomp/libseccomp && git checkout $(LIBSECCOMP_COMMIT) && ./autogen.sh && ./configure --prefix=/usr && make all && sudo make install
```
Comment (mheon) [inline Makefile]: Need to define $LIBSECCOMP_COMMIT somewhere - I suggest whatever's the most recent stable tag?

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**